tests/chaos/oadp/conftest.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+import logging
+
+import pytest
+
+from utilities.constants import TIMEOUT_3MIN, TIMEOUT_5MIN
+from utilities.infra import ExecCommandOnPod, wait_for_node_status
+from utilities.oadp import (
+    FILE_NAME_FOR_BACKUP,
+    TEXT_TO_TEST,
+    VeleroBackup,
+    create_rhel_vm,
+    is_storage_class_support_volume_mode,
+)
+from utilities.storage import write_file
+
+LOGGER = logging.getLogger(__name__)
+
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--volume-mode", action="store", help="Run only tests with the specified volume mode (e.g. block, filesystem)"
+    )
+
+
+def pytest_collection_modifyitems(config, items):
+    selected_mode = config.getoption("--volume-mode")
+    if not selected_mode:
+        return  # No filtering, run all tests
+
+    selected = []
+    deselected = []
+    for item in items:
+        marker = item.get_closest_marker(name="volume_mode")
+        if marker and marker.args and marker.args[0] == selected_mode:
+            selected.append(item)
+        else:
+            deselected.append(item)
+    if deselected:
+        config.hook.pytest_deselected(items=deselected)
+        items[:] = selected
+
+
+def pytest_runtest_setup(item):
+    """
+    Before running each test, check if the StorageClass supports the required volume_mode.
+    Skip the test if not supported.
+    """
+    marker = item.get_closest_marker(name="volume_mode")
+    if not marker:
+        return
+
+    mode = marker.args[0]
+
+    if "snapshot_storage_class_name_scope_module" not in item.fixturenames:
+        return  # Fixture not in the test, skip the check
+
+    sc_name = item._request.getfixturevalue(fixture_name="snapshot_storage_class_name_scope_module")
+
+    if not is_storage_class_support_volume_mode(
+        storage_class_name=sc_name,
+        requested_volume_mode=mode,
+    ):
+        pytest.skip(f"Storage class {sc_name!r} doesn't support volume mode {mode!r}")
```

Comment:
If we explicitly requested a `volume_mode`, but the storage does not support it, the test should fail, not skip